internal/distributed/proxy/httpserver/handler_v1.go

```diff
@@ -0,0 +1,582 @@
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	coll, err := h.describeCollection(c, httpReq.DbName, httpReq.CollectionName, false)
+	if err != nil || coll == nil {
+		return
+	}
+	body, _ := c.Get(gin.BodyBytesKey)
+	err = checkAndSetData(string(body.([]byte)), coll, &httpReq)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "checkout your params", HTTPReturnError: err.Error()})
+		return
+	}
+	req := milvuspb.InsertRequest{
+		CollectionName: httpReq.CollectionName,
+		PartitionName:  "_default",
+		NumRows:        uint32(len(httpReq.Data)),
+	}
+	req.FieldsData, err = anyToColumns(httpReq.Data, coll.Schema)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "insert data by column wrong", HTTPReturnError: err.Error()})
+		return
+	}
+	username, _ := c.Get(ContextUsername)
+	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
+	if authErr != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusUnauthorized, HTTPReturnMessage: authErr.Error()})
+		return
+	}
+	response, err := h.proxy.Insert(c, &req)
+	if err != nil {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "insert fail", HTTPReturnError: err.Error()})
+	} else if response.Status.ErrorCode != commonpb.ErrorCode_Success {
+		c.JSON(http.StatusOK, gin.H{HTTPReturnCode: response.Status.ErrorCode, HTTPReturnMessage: "insert fail", HTTPReturnError: response.Status.Reason})
+	} else {
+		switch response.IDs.GetIdField().(type) {
+		case *schemapb.IDs_IntId:
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: gin.H{"insertCount": response.InsertCnt, "insertIds": response.IDs.IdField.(*schemapb.IDs_IntId).IntId.Data}})
+		case *schemapb.IDs_StrId:
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusOK, HTTPReturnData: gin.H{"insertCount": response.InsertCnt, "insertIds": response.IDs.IdField.(*schemapb.IDs_StrId).StrId.Data}})
+		default:
+			c.JSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "ids' type neither int or string"})
+		}
+	}
+}
+
+func (h *Handlers) search(c *gin.Context) {
+	httpReq := SearchReq{
+		DbName: DefaultDbName,
+		Limit:  100,
+	}
+	if err := c.ShouldBindBodyWith(&httpReq, binding.JSON); err != nil {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "check your parameters conform to the json format", HTTPReturnError: err.Error()})
+		return
+	}
+	if httpReq.CollectionName == "" {
+		c.AbortWithStatusJSON(http.StatusOK, gin.H{HTTPReturnCode: http.StatusBadRequest, HTTPReturnMessage: "collectionName is required."})
+		return
+	}
+	coll, err := h.describeCollection(c, httpReq.DbName, httpReq.CollectionName, false)
```

Comment:
Describing the Collection is unnecessary; the Result contains all the data you require.

Reply:
need to find the index of vector field